src/integrator.jl

```diff
@@ -218,7 +218,7 @@ function step(
     res = if FullTraj
         Vector{P}(undef, n_steps)
     else
-        z
+        Vector{P}(undef, 1)
```

Comment:
Why was this changed? This creates unnecessary allocations in every `step`. There's no type inference issue here since `FullTraj` is a type parameter of a `Val`, so the compiler compiles and optimizes separate methods for `FullTraj = true` and `FullTraj = false` where these branches are eliminated.